modules/azure/keyvault.go

```diff
@@ -0,0 +1,182 @@
+package azure
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	kvauth "github.com/Azure/azure-sdk-for-go/services/keyvault/auth"
+	kvmng "github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2016-10-01/keyvault"
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/stretchr/testify/require"
+)
+
+// KeyVaultSecretExists indicates whether a key vault secret exists; otherwise false
+func KeyVaultSecretExists(t *testing.T, keyVaultName string, secretName string) bool {
+	result, err := KeyVaultSecretExistsE(keyVaultName, secretName)
+	require.NoError(t, err)
+	return result
+}
+
+// KeyVaultKeyExists indicates whether a key vault key exists; otherwise false.  This function would fail the test if there is an error.
+func KeyVaultKeyExists(t *testing.T, keyVaultName string, keyName string) bool {
+	result, err := KeyVaultKeyExistsE(keyVaultName, keyName)
+	require.NoError(t, err)
+	return result
+}
+
+// KeyVaultCertificateExists indicates whether a key vault certificate exists; otherwise false.
+// This function would fail the test if there is an error.
+func KeyVaultCertificateExists(t *testing.T, keyVaultName string, certificateName string) bool {
+	result, err := KeyVaultCertificateExistsE(keyVaultName, certificateName)
+	require.NoError(t, err)
+	return result
+}
+
+// KeyVaultCertificateExistsE indicates whether a certificate exists in key vault; otherwise false.
+// This function would fail the test if there is an error.
+func KeyVaultCertificateExistsE(keyVaultName, certificateName string) (bool, error) {
+	keyVaultSuffix, err := GetKeyVaultURISuffixE()
+	if err != nil {
+		return false, err
+	}
+	client, err := GetKeyVaultClientE()
+	if err != nil {
+		return false, err
+	}
+	maxVersionsCount := int32(1)
+	versions, err := client.GetCertificateVersions(context.Background(),
+		fmt.Sprintf("https://%s.%s", keyVaultName, keyVaultSuffix),
+		certificateName,
+		&maxVersionsCount)
+	if err != nil {
+		return false, err
+	}
+
+	if len(versions.Values()) > 0 {
+		return true, nil
+	}
+	return false, errors.New("No certificate versions found")
```

Comment:
I think this should be `false, nil`. Otherwise, there is no path where this function returns `false` normally, which means that `KeyVaultCertificateExists` (the non-E version) would only return`true`, or fail the test, which isn't what you would expect to happen for `Exists` function.

Alternatively, we can rename the function to `AssertKeyVaultCertificateExists`.

This applies to all the new functions here.

Reply:
@yorinasub17 I agree. The no versions case equals a non existing secret/key/certificate. Updated the PR.